File: obelisk_ws/src/python/obelisk_sim_py/obelisk_sim_py/obelisk_mujoco_robot.py
from typing import List, Optional

from obelisk_control_msgs.msg import PositionSetpoint
from rclpy.executors import MultiThreadedExecutor

from obelisk_py.core.sim.mujoco import ObeliskMujocoRobot
from obelisk_py.core.utils.ros import spin_obelisk


def main(args: Optional[List] = None) -> None:
    """Main entrypoint."""
    ctrl_msg_type = PositionSetpoint
    spin_obelisk(args, ObeliskMujocoRobot, MultiThreadedExecutor, {"ctrl_msg_type": ctrl_msg_type})


if __name__ == "__main__":
    main()

File: obelisk_ws/src/cpp/obelisk_sim_cpp/src/obelisk_mujoco_robot.cpp
#include "rclcpp/rclcpp.hpp"

#include "obelisk_mujoco_sim_robot.h"
#include "obelisk_ros_utils.h"

int main(int argc, char* argv[]) {
    obelisk::utils::SpinObelisk<obelisk::ObeliskMujocoRobot<obelisk_control_msgs::msg::PositionSetpoint>,
                                rclcpp::executors::MultiThreadedExecutor>(argc, argv, "mujoco_sim");
}
